File: bench/cross-language/primes_sieve/primes_sieve.ts
// Sieve of Eratosthenes — primes up to 1,000,000

const limit = 1000000;
const sieve = new Uint8Array(limit + 1);

sieve[0] = 1;
sieve[1] = 1;

for (let i = 2; i * i <= limit; i++) {
  if (!sieve[i]) {
    for (let j = i * i; j <= limit; j += i) {
      sieve[j] = 1;
    }
  }
}

let count = 0;
for (let i = 2; i <= limit; i++) {
  if (!sieve[i]) count++;
}

console.log(`primes_sieve(1000000): count = ${count}`);

File: bench/cross-language/primes_sieve/primes_sieve.rs
// Sieve of Eratosthenes — primes up to 1,000,000

fn main() {
    let limit: usize = 1_000_000;
    let mut sieve = vec![false; limit + 1];
    sieve[0] = true;
    sieve[1] = true;

    let mut i = 2;
    while i * i <= limit {
        if !sieve[i] {
            let mut j = i * i;
            while j <= limit {
                sieve[j] = true;
                j += i;
            }
        }
        i += 1;
    }

    let count = sieve[2..].iter().filter(|&&x| !x).count();
    println!("primes_sieve(1000000): count = {}", count);
}
